staging/src/k8s.io/sample-controller/docs/controller-client-go.md

```diff
@@ -0,0 +1,59 @@
+# client-go under the hood
+
+The [client-go](https://github.com/kubernetes/client-go/) library contains various mechanisms that you can use when 
+developing your custom controllers. These mechanisms are defined in the 
+[tools/cache folder](https://github.com/kubernetes/client-go/tree/master/tools/cache) of the library.
+
+Here is a pictorial representation showing how the various components in 
+the client-go library work and their interaction points with the custom 
+controller code that you will write.
+
+<p align="center">
+  <img src="images/client-go-controller-interaction.jpeg" height="600" width="700"/>
+</p>
+
+## client-go components
+
+* Reflector: A reflector [watches the Kubernetes API](https://github.com/kubernetes/client-go/blob/master/tools/cache/reflector.go#L239) for the specified 
```

Comment:
these links will unfortunately break easily. Can we mention the types with their package instead? That's more stable than line numbers.

Reply:
Ok, makes sense. Will do that and resubmit.